packages/ckeditor5-engine/src/view/document.ts

```diff
@@ -96,6 +97,19 @@ export default class Document extends BubblingEmitterMixin( ObservableMixin() )
 		 */
 		this.set( 'isFocused', false );
 
+		/**
+		 * True if document is changing the focus.
+		 *
+		 * This property has been introduced due to the different behaviors on browsers.
+		 * Chromium sends the focus event before the changeselection which causes the renders with the old selection.
+		 * This flag is used to prevent rerenders when the focus is in progress and we are still waiting for new value.
+		 *
+		 * @readonly
+		 * @observable
+		 * @member {Boolean} module:engine/view/document~Document#isFocusChanging
+		 */
+		this.set( 'isFocusChanging', false );
```

Comment:
```suggestion
		/**
		 * Set to `true` if the document is in the process of setting the focus.
		 *
		 * To be precise, there are two browser events that we care about: `focus` and `selectionchange`. 
		 *
		 * Different browsers handle them differently. Chromium sends the `focus` event before the
		 * `selectionchange` event, which leads to rendering with the old selection state.
		 *
		 * The flag is used to prevent rendering when setting the focus is in progress
		 * and we are waiting for the new value.
		 *
		 * @readonly
		 * @observable
		 * @member {Boolean} module:engine/view/document~Document#isFocusChanging
		 */
		this.set( 'isFocusChanging', false );
```